### PR TITLE
Small improvements

### PR DESCRIPTION
**Pull request by JanLoebel**
Improvements:
- Updated .gitignore to ignore eclipse-projects
- Updated version of spring-boot to 1.5.16
- Refactored classes to be better testable
- Assured that a typo is really guaranteed

### CODE REVIEW

File: src/main/java/com/github/jolangbein/RandomTypoGenerator.java [hunk 1]
```diff
@@ -10,7 +11,7 @@
     private static final int HIGHEST_ASCII_CHAR = 126;
     private static final int LOWEST_ASCI_CHAR = 33;
 
-    private Random random = new Random();
+    private final Random random = new SecureRandom();
```
Comment (joc59) [inline src/main/java/com/github/jolangbein/RandomTypoGenerator.java]: Nice! Generating typos cannot be taken lightly, so we must use cryptographcally secure random numbers here. Good thinking! Would you like to update the README file so we can brag about this more publicly?

**Review by joc59 — COMMENTED** [review]
We definitely need some serious code formatting rules. I'll schedule us a meeting of approx. 6 hours

File: src/main/java/com/github/jolangbein/RandomTypoGenerator.java (hunk 1 above)
Comment (JanLoebel) [reply src/main/java/com/github/jolangbein/RandomTypoGenerator.java]: Not sure if we should brag about it. In my opinion, everybody deserves the right/possibility/freedom to have secure typos by secure random numbers!

**Comment by JanLoebel** [comment]
Sounds great, but I'm pretty sure we should include a lot of people in that discussion to have many different optionions about tabs/spaces or where to put the curly brackets. Everybody should prepare a presentation about his favorite style guide and we shoud'nt vote on it, we should let everybody discuss it till finally everybody want the same style!

**Pull request merged**